bundles/org.openhab.ui/web/src/components/widgets/chart/series/oh-mark-area.js

```diff
@@ -1,20 +1,32 @@
+import dayjs from 'dayjs'
+import LocalizedFormat from 'dayjs/plugin/localizedFormat'
+dayjs.extend(LocalizedFormat)
+
 export default {
-  get (component, points) {
+  get (component, points, startTime, endTime, chart) {
     const itemPoints = points.find(p => p.name === component.config.item).data
     let markArea = Object.assign({}, component.config)
     markArea.data = []
     let rollingState = null
     let currentArea = null
     itemPoints.forEach((p) => {
-      if (p.state === 'ON' && rollingState !== 'ON') {
+      if ((p.state === 'ON' && rollingState !== 'ON') || (p.state === 'OPEN' && rollingState !== 'OPEN')) {
```

Comment:
Should state constants like OPEN and ON be centralized somewhere ?  More of a general question than a suggestion to change anything.

Reply:
It's quite early and experimental for now, but I'd like to improve the handling of discrete values in general. For example look up in the state description and consider even numbers a as discrete series if options are present (e.g. scene numbers, weather conditions and the like) - if multiple discrete series are to be displayed at the same time, the areas could overlap so it could also be rendered as colored bars, like the timeline control in HABPanel; tooltips and the like would display the relevant label if found in the state description options.